src/sql/workbench/contrib/commandLine/electron-browser/commandLine.ts

```diff
@@ -324,9 +368,31 @@ export class CommandLineWorkbenchContribution implements IWorkbenchContribution,
 		profile.setOptionValue('applicationName', applicationName);
 		profile.setOptionValue('databaseDisplayName', profile.databaseName);
 		profile.setOptionValue('groupId', profile.groupId);
+		// Set all advanced options
+		let advancedOptions = this.getAdvancedOptions(args.properties, profile.getOptionKeyIdNames());
+		advancedOptions.forEach((v, k) => {
+			profile.setOptionValue(k, v);
+		});
 		return this._connectionManagementService ? this.tryMatchSavedProfile(profile) : profile;
 	}
 
+	private getAdvancedOptions(options: string, idNames: string[]): Map<string, string> {
+		const ignoredProperties = idNames.concat(['password', 'azureAccountToken']);
+		let advancedOptionsMap = new Map<string, string>();
+		if (options) {
+			try {
+				JSON.parse(options, (k, v) => {
+					if (!(k in ignoredProperties)) {
+						advancedOptionsMap.set(k, v);
+					}
+				});
+			} catch (e) {
+				throw new Error(localize('commandline.propertiesFormatError', 'Advanced connection properties could be parsed as JSON. Received properties value: {0}', options));
```

Comment:
I would also suggest adding the original error message in so the user can see more information about what the specific parsing failure was (or log it and tell the user they can see the logs for more information)

```suggestion
				throw new Error(localize('commandline.propertiesFormatError', 'Advanced connection properties could not be parsed as JSON. Received properties value: {0}', options));
```